prov/rxd/src/rxd_cntr.c

```diff
@@ -60,7 +60,7 @@ static int rxd_cntr_wait(struct fid_cntr *cntr_fid, uint64_t threshold, int time
 			return -FI_ETIMEDOUT;
 
 		ep_retry = -1;
-		ofi_genlock_lock(&cntr->ep_list_lock);
+		ofi_mutex_lock(&cntr->ep_list_lock);
```

Comment:
Same thing here - they are the same thing, I don't think this should affect performance. We should take a closer look at it.

Reply:
they are not the same thing, and they do impact performance.

Reply:
nope... please double check my testing.